src/typescript/azure-openapi-validator/rules/OperationIdRequired.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+import { isValidOperation } from './utilities/rules-helper';
+export const OperationIdRequired: string = "OperationIdRequired";
+
+rules.push({
+  id: "R4004",
+  name: OperationIdRequired,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
+  appliesTo_JsonQuery: "$.paths",
+  run: function* (doc, node, path) {
+    if (node !== undefined) {
+      for (const pathKey in node)
+        for (const op in node[pathKey]) {
+          if (!isValidOperation(op)) {
```

Comment:
Could you combine those 2 'if' together?

Reply:
I prefer remain it which will be better understanding